### DISCUSSION TITLE
Handling Algorand Clear State

### DISCUSSION THREAD

**Discussion by No-Cash-7970** [reach-sh/reach-lang#1267] [Q&A] [answered]
For Algorand, is there a way to execute code when a user opts out of the contract in Reach, like Algorand's `ClearState` handler? 

The reason for this is that I have a global counter of the number of users opted into the contract. So, when a user opts out of the contract, I want that counter to be updated.

**Comment by jeapostrophe** [comment, marked as answer]
We don't support this now. I am anxious about adding it because it will extremely expensive in space, because you'll effectively need to duplicate the entire state-decoding part of the program. Very soon, Algorand will have box storage, where this is not an issue, because the application has complete authority over the storage space.
